### ISSUE TITLE
Feature request: PDF upload

### ISSUE THREAD

**Issue by jsabol** [jvsteiner/s3-image-uploader#2]
Hi!! LOVING the plugin so far - this is flawless and immensely useful. I was wondering if we could get support for other file formats, like pdfs, etc. I have a lot of pdfs and it would be great to embed them directly and upload to s3! Maybe a configurable list of file formats we can upload? 

Thanks so much for this plugin, it's incredible

**Comment by jvsteiner** [comment]
Hi, thanks for trying it out.  Rendering pdfs is probably outside the scope of what I was trying to achieve with this - there is another plugin I use for that - "Ozan's Image in Editor Plugin".  You can use it to transclude pdf's - maybe you are already using it, or something similar.

I have understood that what you are trying to do would be something like that, but with the file uploaded.

I tried it, and unfortunately, while it was easy to get the file to upload, I wasn't able to find a way to make transclusion of the resulting uploaded file work - at least not with Ozan's.  A trivial problem is that the link style is different: for pdf'susing Ozan's it would be ![[file]]  vs. ![name](file) - but even if you change the style manually, Ozan's doesn't transclude the remote file.  

In short, I think Ozan's would need to make a change to allow transclusion of remote links for these two plugins to play well.

Do you know of a pdf transclusion plugin that works with remote links?

**Comment by jsabol** [comment]
You're right! Idk why I thought an image markdown syntax would work.

**Comment by jsabol** [comment]
You're right! idk why I thought image syntax would work 

I tried using an iframe, but unfortunately it doesn't work. The url is blocked in the Network panel. There's a feature request to fix this: https://forum.obsidian.md/t/iframing-external-pdf-files-currently-empty-element-loaded/26884

I did find a trick though from the discord!

With a url of http://julietfoundryvtt.s3.amazonaws.com/art/obsidian/Heroes-Handbook-5E_635ec2ff48843.pdf, we can use the google docs viewer and it will iframe as expected.

```
<iframe frameborder=0 border=0 width=100% height=800 src="https://docs.google.com/viewer?url=http://julietfoundryvtt.s3.amazonaws.com/art/obsidian/Heroes-Handbook-5E_635ec2ff48843.pdf?raw=true"></iframe>
```

I tested this in a vault with no plugins and it worked great. As far as Ozan's, it looks like it will embed the pdfs, but only when you have Live Preview off (boooo) haha. The iframe that gets embedded by Ozan's when I inspect the html doesn't work with Ozan's off. No idea what flip they're switching to let those urls through.

**Comment by jsabol** [comment]
External audio and video files can be included too: 

```html
<audio src="http://julietfoundryvtt.s3.amazonaws.com/music/voice-record/holyfuck.mp3" controls />
<video src="http://julietfoundryvtt.s3.amazonaws.com/maps/animated/Sewer_Set2_T_intersection_WNE%20HD_GRID.mp4" controls />
```

Maaaaan, would this change the game. (I think uploading these huge files would necessitate some kind of "Loading..." placeholder text while it's going, but I can definitely live with just waiting and knowing it's probably working)

**Comment by jvsteiner** [comment]
I'll give it a try

**Comment by jvsteiner** [comment]
Does the video tag work for you?  I cant get the video to load...

**Comment by jvsteiner** [comment]
added in 0.2.3

**Comment by jsabol** [comment]
Omg this is amazing!!!! Holy shit I can't believe how quickly you turned that around. Video, audio, and pdfs are all working for me! Thank you. Do you have a way I can send some cash your way?

**Comment by jvsteiner** [comment]
TYVM!
